pika/adapters/base_connection.py

```diff
@@ -239,7 +241,10 @@ def _get_error_code(self, error_value):
         return None
 
     def _flush_outbound(self):
-        """Call the state manager who will figure out that we need to write."""
+        """write early, if the socket will take the data why not get it out
+        there asap.
+        """
+        self._handle_write()
```

Comment:
When tornado ioloop calls `BaseConnection._handle_events` and READ is indicated, it calls `BaseConnection._handle_read`, which will attempt to read by default at most `pika.spec.FRAME_MAX_SIZE = 131072` bytes from the socket and pass that to `Connection._on_data_available`. The pika overhead of processing this data should be pretty minimal, so processing time of incoming messages largely depends on processing by user code.